Framework/Kernel/src/DiskBuffer.cpp

```diff
@@ -52,19 +52,18 @@ void DiskBuffer::toWrite(ISaveable *item) {
                              // its size in memory
   {
     // forget old memory size
-    m_mutex.lock();
+    std::unique_lock<std::mutex> uniqueLock(m_mutex);
```

Comment:
The `std::lock_guard` cannot be unlocked manually, the `std::unique_lock` can. In the original code mutex needs to be unlocked before `item->setBufferSize(newMemorySize);` line.